parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -254,11 +254,7 @@ if [[ -n ${MASTER_NODE} ]]; then
     time_metric "EnsureEtcd" ensureEtcd
   fi
   time_metric "EnsureK8sControlPlane" ensureK8sControlPlane
-  if [ -f /var/run/reboot-required ]; then
-    time_metric "ReplaceAddonsInit" replaceAddonsInit
-  else
-    time_metric "EnsureAddons" ensureAddons
-  fi
+  time_metric "EnsureAddons" ensureAddons
```

Comment:
Again, addons are a control plane-enforceable bit of foo, so we want to do that now always. If we skip this in a "will reboot later" scenario, that means we will not perform any addons validation at all (CSE does not re-run after a reboot).